### ISSUE TITLE
Can't export in odf/pdf/doc

### ISSUE THREAD

**Issue by manu-p** [ether/etherpad-lite#4035]
Hi,
**Etherpad version**

- Version number: 1.8.3
- Git sha: 6210114

**Installed plugins**

- ep_author_hover
- ep_delete_empty_pads
- ep_etherpad-lite
- ep_font_size
- ep_font_color

On a **specific** pad, **not all**, trying to export the pad in odf, pdf or even doc fails. 

A new tab is opened remaining white for a while and then displays a 502 error:
**Proxy Error**
The proxy server received an invalid response from an upstream server.
The proxy server could not handle the request
Reason: Error reading from remote server
Apache/2.4.29 (Ubuntu) Server at pad.domain.fr Port 443

The Etherpad-lit access log shows:
```
ip@ - - [19/May/2020:13:51:16 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8inWo2&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:51:41 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8incug&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:51:20 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8inXlb&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:51:46 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8ine4R&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:51:41 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8incw9&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:52:06 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8inj0n&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:52:12 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8inkQg&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:51:46 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8ine5s&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:52:06 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8inj2U&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:52:31 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8inp97&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:52:12 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8inkS1&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:52:38 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8inqmw&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:52:31 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8inpAh&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:52:56 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8invHJ&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:53:04 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8inx7A&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:52:38 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8inqoL&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:52:56 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8invIs&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:53:21 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8in_PV&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:53:04 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8inx8b&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:53:30 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8io1TQ&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:51:39 +0200] "GET /p/gdjrxz43hodtnuhz/export/pdf HTTP/1.1" 502 1296 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:53:21 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8in_Qz&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:53:46 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8io5Xc&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:53:30 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8io1Up&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:53:56 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8io7pg&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:54:12 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8ioBfk&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:53:46 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8io5Z5&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:53:56 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8io7r4&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:54:22 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8ioE9w&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:54:12 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8ioBhD&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:54:37 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8ioHns&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:54:22 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8ioEBI&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:54:48 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8ioKWB&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:55:02 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8ioNw6&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:54:37 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8ioHpV&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:54:48 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8ioKXb&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:55:14 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8ioQsQ&sid=QvT5PxOf34dr8-ZkAACG HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
ip@ - - [19/May/2020:13:55:02 +0200] "GET /socket.io/?EIO=3&transport=polling&t=N8ioNxe&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 305 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
ip@ - - [19/May/2020:13:55:27 +0200] "POST /socket.io/?EIO=3&transport=polling&t=N8ioU2H&sid=vmTd850wyln4FD54AACH HTTP/1.1" 200 938 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0"
```
And the Etherpad-lite error log:
```

[Tue May 19 13:53:39.712825 2020] [proxy_http:error] [pid 11560] (20014)Internal error (specific information not available): [client ip@:57544] AH01102: error reading status line from remote server 127.0.0.1:9001
[Tue May 19 13:53:39.713080 2020] [proxy:error] [pid 11560] [client ip@:57544] AH00898: Error reading from remote server returned by /p/gdjrxz43hodtnuhz/export/pdf
```
It's quite rare, to be honest the very 1st time I see it, but remains unexplained so far...

**Comment by JohnMcLear** [comment]
Please export the ``.etherpad`` of ``gdjrxz43hodtnuhz`` and paste the contents here or online.

Does the export fail if you disable plugins?

**Comment by manu-p** [comment]
Thanks @JohnMcLear 

Where do I find the `.etherpad` of `gdjrxz43hodtnuhz`?

This is a production system, I can't disable plugins during working hours.

**Comment by JohnMcLear** [comment]
http://instance/p/gdjrxz43hodtnuhz/export/etherpad

**Comment by manu-p** [comment]
I'm afraid the file is too big to be pasted here, you should be able to download it from [here](https://cloud.combustible-numerique.fr/s/mAXtz32af5JHGcz).

**Comment by JohnMcLear** [comment]
I can export in those formats fine.

Steps to isolate problem.

1. Create local dev environment replicating production (this is **always** good practice)
1. Remove plugins, test.
1. If plugin fails, create issue on that plugins repo (font family has known issue already afaik).
1. If you are using abiword try change to soffice, it's more reliable imho.
1. Once completed all steps above come back.

I think this is not an issue with Etherpad though because I can export the file on my dev instance :)

**Comment by manu-p** [comment]
Yep, that's what I was doing in the mean time.

So I imported this etherpad on another instance (on nginx this time), same version number: 1.8.3 (Git sha: 82d1179) and **no plugin** at all, and **abiword** as for the 1st instance.
And still the same export failure...

So the next step is to use soffice instead of abiword, and I installed LibreOffice Writer on this 2nd server, modified settings.json so that `"abiword" : null,` and `"soffice" : "/usr/bin/soffice",`.
I noticed that `"tidyHtml" : "/usr/bin/tidy",` so I reverted it back to null, in case...

Then re-opened the pad and tried to export it again in odf or pdf with no better result.

You say you're able to export it in odf, are you using the same Etherpad-Lite than ours?
